commit 0671cc1048744c9a6f1c896baa85966a5abc42a0
Author: Harald Freudenberger <freude@linux.ibm.com>
Date:   Mon Jul 27 14:34:57 2020 +0200

    s390/sclp: Add support for SCLP AP adapter config/deconfig
    
    Add support for AP bus adapter config and deconfig to the sclp
    core code. The code is statically build into the kernel when
    ZCRYPT is configured either as module or with static support.
    
    This is the base functionality for having configure/deconfigure
    support in the AP bus and card code. Another patch will exploit
    this soon.
    
    Signed-off-by: Harald Freudenberger <freude@linux.ibm.com>
    Suggested-by: Pierre Morel <pmorel@linux.ibm.com>
    Signed-off-by: Vasily Gorbik <gor@linux.ibm.com>

diff --git a/arch/s390/include/asm/sclp.h b/arch/s390/include/asm/sclp.h
index 90f34c7e2752..a7bdd128d85b 100644
--- a/arch/s390/include/asm/sclp.h
+++ b/arch/s390/include/asm/sclp.h
@@ -128,6 +128,8 @@ int sclp_chp_deconfigure(struct chp_id chpid);
 int sclp_chp_read_info(struct sclp_chp_info *info);
 int sclp_pci_configure(u32 fid);
 int sclp_pci_deconfigure(u32 fid);
+int sclp_ap_configure(u32 apid);
+int sclp_ap_deconfigure(u32 apid);
 int sclp_pci_report(struct zpci_report_error_header *report, u32 fh, u32 fid);
 int memcpy_hsa_kernel(void *dest, unsigned long src, size_t count);
 int memcpy_hsa_user(void __user *dest, unsigned long src, size_t count);